docs/concepts/json_schema.md

```diff
@@ -572,6 +725,63 @@ print(m.model_fields)
 """
 ```
 
+### `WithJsonSchema` annotation
+
+??? api "API Documentation"
+    [`pydantic.json_schema.WithJsonSchema`][pydantic.json_schema.WithJsonSchema]<br>
+
+The [`WithJsonSchema`][pydantic.json_schema.WithJsonSchema] annotation can be used to override the generated (base)
+JSON schema for a given type without the need to implement `__get_pydantic_core_schema__` on the type itself.
+
+This provides a way to set a JSON schema for types that would otherwise raise errors when producing a JSON schema,
+such as `Callable`, or types that have an `is-instance` core schema.
+
+For example, the use of a [`PlainValidator`][pydantic.functional_validators.PlainValidator] in the following example
+would otherwise raise an error when producing a JSON schema because the [`PlainValidator`][pydantic.functional_validators.PlainValidator]
+is a `Callable`. However, by using the [`WithJsonSchema`][pydantic.json_schema.WithJsonSchema]
+annotation, we can override the generated JSON schema for the custom `MyInt` type:
+
+```py
+from typing_extensions import Annotated
+
+from pydantic import BaseModel, PlainValidator, WithJsonSchema
+
+MyInt = Annotated[
+    int, PlainValidator(lambda v: int(v) + 1), WithJsonSchema({'type': 'int'})
+]
+
+
+class Model(BaseModel):
+    a: MyInt
+
+
+print(Model(a='1').a)
+#> 2
+
+print(Model(a='1').model_json_schema())
+"""
+{
+    'properties': {'a': {'title': 'A', 'type': 'int'}},
+    'required': ['a'],
+    'title': 'Model',
+    'type': 'object',
+}
+"""
+```
+
+In the future, it's likely that Pydantic will add builtin support for JSON schema generation for types like
+[`PlainValidator`][pydantic.functional_validators.PlainValidator], but the
+[`WithJsonSchema`][pydantic.json_schema.WithJsonSchema] annotation will still be useful for other custom types.
```

Comment:
Is there an issue or discussion where we talk about this?

If so, we can add it.

Reply:
Good question! There's this one: https://github.com/pydantic/pydantic/issues/7957. I'll add a link.

Reply:
Is there any open issue? What I meant is to be able to track the "future" mentioned here.

Reply:
Not that I know of. I can make one...